src/core/stream.c

```diff
@@ -44,6 +44,36 @@ static struct {
 	    .listener_alloc = nni_ipc_listener_alloc,
 	},
 #endif
```

Comment:
I do not quite understand what you want exactly ..., perhaps just rename it? I prefer to let users share same entrance just as SP when using nng for mqtt.

Reply:
The issue here is that these URLs describe *transports* (stream protocols).  You should not declare new ones here.

MQTT should for example just use tcp:// or tls+tcp:// or whatever in its URLs.  No need for an MQTT prefix.

The bigger problem here is that you really shouldn't be using the low level SP listener and dialer framework.  (In fact, for MQTT, being a listener is non-sensical since only an MQTT broker should listen.)

I've been working on moving the listener/dialer/pipe/socket semantics that are SP specific into the SP directory, and exposing only some very very high level common semantics (using an object oriented design) at the common level.

Reply:
Well, MQTT requires an independent transports protocol as well. I believe we discussed this before, about how to implement QoS while keep zero-copy of nng.
Anyway, what would you want me to do with URL? How about I duplicate the whole set of listener and dialer framework for mqtt first?